similarity-data-impl/src/main/java/org/phenotips/data/similarity/internal/RestrictedDisorderSimilarityView.java

```diff
@@ -52,73 +45,31 @@
      */
     public RestrictedDisorderSimilarityView(Disorder match, Disorder reference, AccessType access)
     {
-        this.match = match;
-        this.reference = reference;
+        super(match, reference);
         this.access = access;
     }
 
     @Override
     public String getId()
     {
-        return this.access.isOpenAccess() && this.match != null ? this.match.getId() : null;
+        return this.access.isPrivateAccess() ? null : super.getId();
```

Comment:
Yep, I think we want to go ahead and disclose the diagnosis for matchable cases.